tools/custom-stats-qc/1.0.0/custom-stats-qc__1.0.0.cwl

```diff
@@ -0,0 +1,154 @@
+          JSON_STRING=\$( jq -n \\
+                            --arg des "pct discordant reads" \\
+                            --arg src "SN	percentage of properly paired reads (%)" \\
+                            --arg val "$pct_discordant_reads" \\
+                            '{pct_discordant_reads: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract mean_insert_size
+          mean_insert_size=\$(grep -h "SN	insert size average:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          JSON_STRING=\$( jq -n \\
+                            --arg des "mean insert size" \\
+                            --arg src "SN	insert size average" \\
+                            --arg val "$mean_insert_size" \\
+                            '{mean_insert_size: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract Insert_size_SD
+          insert_size_sd=\$(grep -h "SN	insert size standard deviation:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          JSON_STRING=\$( jq -n \\
+                            --arg des "insert size sd" \\
+                            --arg src "SN	insert size standard deviation" \\
+                            --arg val "$insert_size_sd" \\
+                            '{insert_size_sd: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          # Extract pct_mapped_reads where pct_mapped_reads = [(Mapped Reads - Reads MQ0) / Total Reads]
+          # RM = Mapped Reads
+          # R0 = Reads MQ0
+          # RT = Total Reads
+          RM=\$(grep -h "SN	reads mapped:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          R0=\$(grep -h "SN	reads MQ0:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          RT=\$(grep -h "SN	raw total sequences:" "$(inputs.output_samtools_stats.path)" | cut -f3)
+          pct_mapped_reads=\$(( 100 * ( $RM - $R0) / $RT ))
+          JSON_STRING=\$( jq -n \\
+                            --arg des "pct mapped reads" \\
+                            --arg src "pct_mapped_reads = [(Mapped Reads - Reads MQ0) / Total Reads]" \\
+                            --arg val "$pct_mapped_reads" \\
+                            '{pct_mapped_reads: {description: $des, source: $src, value: $val}}' )
+          echo \$JSON_STRING >> "$(inputs.output_json_filename).json"
+
+          jq '.' "$(inputs.output_json_filename).json" >> "$(inputs.output_json_filename)_$(inputs.sample_id).json"
+
+baseCommand: [ "sh", "run-custom-qc.sh" ]
+
+inputs:
+  sample_id:
+    label: sample id
+    doc: |
+      Id of the input sample
+    type: string
+  output_samtools_stats:
+    label: output samtools stats
+    doc: |
+      Output txt file from samtools stats
+    type: File
+  output_json_filename:
```

Comment:
Being a bit pedantic here, but if we're pulling out `output_json_filename + sample_id` should `output_json_filename` just be `output_json_prefix`?

Reply:
Wasn't required at the end as went with a single output file https://github.com/umccr/cwl-ica/blob/52ee1dd285bbb8d113e4e53f9dbf23c38cb9c750/tools/custom-stats-qc/1.0.1/custom-stats-qc__1.0.1.cwl#L150

See https://github.com/umccr/cwl-ica/blob/52ee1dd285bbb8d113e4e53f9dbf23c38cb9c750/tools/custom-stats-qc/1.0.1/custom-stats-qc__1.0.1.cwl#L117